### PR TITLE
[MNG-7520] Simplify integration of Redisson and Hazelcast for Maven R…

### PR DESCRIPTION
**Pull request by michael-o**
…esolver

This closes #776

Following this checklist to help us incorporate your
contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MNG) filed
       for the change (usually before you start working on it).  Trivial changes like typos do not
       require a JIRA issue. Your pull request should address just this issue, without
       pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[MNG-XXX] SUMMARY`, where you replace `MNG-XXX`
       and `SUMMARY` with the appropriate JIRA issue. Best practice is to use the JIRA issue
       title in the pull request title and in the first line of the commit message.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will
       be performed on your pull request automatically.
 - [ ] You have run the [Core IT][core-its] successfully.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [ ] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

[core-its]: https://maven.apache.org/core-its/core-it-suite/

### CODE REVIEW

**Comment by michael-o** [comment]
@jebeaudet have a look

**Comment by cstamas** [comment]
Am unsure about the "worthiness" of this PR.... I mean, will someone really want to use Hazelcast AND Redisson? 
IMHO, this is -0 as am pretty sure people will "stick" to one or another, and also, there is file locking...

**Comment by michael-o** [comment]
> Am unsure about the "worthiness" of this PR.... I mean, will someone really want to use Hazelcast AND Redisson? IMHO, this is -0 as am pretty sure people will "stick" to one or another, and also, there is file locking...

It is not about having BOTH in place. It shalls imply reduce manual setup and simplification of our documentation. Both are contained to handle both cases, NOT at the same time.

**Comment by michael-o** [comment]
Note that this does not include any JARs, just the skeleton to make it happen.

File: apache-maven/src/assembly/maven/bin/m2.conf [hunk 1]
```diff
@@ -4,5 +4,7 @@ set maven.conf default ${maven.home}/conf
 
 [plexus.core]
 load       ${maven.conf}/logging
+optionally ${maven.home}/lib/ext/redisson/*.jar
```
Comment (jebeaudet) [inline apache-maven/src/assembly/maven/bin/m2.conf:7]: nice!

File: apache-maven/src/assembly/maven/lib/ext/hazelcast/README.txt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+This directory is intended to contain Hazelcast JARs for Maven Resolver Named Locks using Hazelcast.
```
Comment (jebeaudet) [inline apache-maven/src/assembly/maven/lib/ext/hazelcast/README.txt:1]: would it be worth it to add a link to redisson/hazelcast github?

**Review by jebeaudet — APPROVED** [review]

File: apache-maven/src/assembly/maven/lib/ext/hazelcast/README.txt (hunk 2 above)
Comment (michael-o) [reply apache-maven/src/assembly/maven/lib/ext/hazelcast/README.txt:1]: Sure, make a proposal.

**Comment by michael-o** [comment]
@cstamas how should I continue here? This is a reasonable simplification for everyone wants to use either one.

**Comment by michael-o** [comment]
@cstamas Is my reasoning clear on this?

**Review by cstamas — APPROVED** [review]

**Pull request closed without merging**